### PR TITLE
Fix many, many compiler warnings

### PR DESCRIPTION
**Pull request by bluca**
This PR is mostly from another colleague, and fixes really a lot of compiler warnings. There will be more in the future, the end goal is to have a full compilation without warnings.

Also a fix for a call in the unit tests, which is not working, and to fix logging of module IDs which would sometimes overflow.

### CODE REVIEW

File: src/common/Time.h [hunk 1]
```diff
@@ -242,9 +242,11 @@ inline uint32_t ntp32_substract (uint32_t x, uint32_t y)
 
 inline void convertNtp64(uint64_t ntptime, uint64_t& result)
 {
-        uint64_t hbnum = ntohll(*(uint64_t*)&ntptime);
+        uint64_t hbnum = ntohll(ntptime);
         if (hbnum>0) {
-                timeval t = timentp64(*((ntp64*)(&hbnum)));
+                ntp64 ntp64_time;
+                memcpy(&ntp64_time, &hbnum, 8);
```
Comment (ogasser) [inline src/common/Time.h]: `sizeof(ntp64)` instead of `8`?

File: src/modules/packet/Packet.h [hunk 2]
```diff
@@ -285,15 +285,18 @@ class Packet :  public ManagedInstance<Packet>, public Emitable
 			transportHeaderOffset = (( *data.netHeader & 0x0f ) << 2);
 
 			// crop layer 2 padding
-			unsigned int endOfIpOffset = layer2HeaderLen +  ntohs(*((uint16_t*) (data.netHeader + 2)));
+			uint16_t ip_total_length;
+			memcpy(&ip_total_length, (data.netHeader+2), 2);
```
Comment (ogasser) [inline src/modules/packet/Packet.h]: Use `sizeof()` in memcpy to increase readability?

File: src/modules/packet/Packet.h [hunk 3]
```diff
@@ -285,15 +285,18 @@ class Packet :  public ManagedInstance<Packet>, public Emitable
...
+			unsigned int endOfIpOffset = layer2HeaderLen +  ntohs(ip_total_length);
 			if(data_length > endOfIpOffset)
 			{
 				DPRINTF("crop layer 2 padding: old: %u  new: %u\n", data_length, endOfIpOffset);
 				data_length = endOfIpOffset;
 			}
 
 			// get fragment offset
-			fragoffset = (*(uint16_t*)(data.netHeader+6))&0xFF1F;
+			memcpy(&fragoffset, (data.netHeader+6), 2);
```
Comment (ogasser) [inline src/modules/packet/Packet.h]: Use `sizeof()` in memcpy to increase readability?

File: src/modules/packet/Packet.h [hunk 4]
```diff
@@ -324,7 +327,9 @@ class Packet :  public ManagedInstance<Packet>, public Emitable
 
 					case 44:	// Fragment
 						// Only use transport header if this is the first fragment
-						fragoffset = ((*((uint16_t*) (data.netHeader + transportHeaderOffset + 2)))) & 0xF8FF;
+						memcpy(&fragoffset, (data.netHeader + transportHeaderOffset + 2), 2);
```
Comment (ogasser) [inline src/modules/packet/Packet.h]: Use `sizeof()` in memcpy to increase readability?

File: src/modules/packet/Packet.h [hunk 5]
```diff
@@ -348,7 +353,9 @@ class Packet :  public ManagedInstance<Packet>, public Emitable
 			}
 
 			// crop layer 2 padding
-			unsigned int endOfIpOffset = layer2HeaderLen +  ntohs(*((uint16_t*) (data.netHeader + 2)));
+			uint16_t ip_total_length;
+			memcpy(&ip_total_length, (data.netHeader+2), 2);
```
Comment (ogasser) [inline src/modules/packet/Packet.h]: Use `sizeof()` in memcpy to increase readability?

File: src/common/ipfixlolib/ipfixlolib.c [hunk 6]
```diff
@@ -1316,9 +1316,6 @@ static int valid_transport_protocol(enum ipfix_transport_protocol p) {
 #ifdef IPFIXLOLIB_RAWDIR_SUPPORT
```
Comment (ogasser) [inline src/common/ipfixlolib/ipfixlolib.c]: Should we move the `#ifdef` after the `case RAWDIR` and leave the `#else` part intact? Then it would be similar to the SUPPORT_SCTP ifdef?

Comment (bluca) [inline src/common/ipfixlolib/ipfixlolib.c]: The problem is that RAWDIR exists only if that define is set:

```
src/common/ipfixlolib/ipfixlolib.c:1316:7: error: ‘RAWDIR’ undeclared (first use in this function)
  case RAWDIR:
```
src/common/ipfixlolib/ipfixlolib.h:

```
enum ipfix_transport_protocol {
#ifdef IPFIXLOLIB_RAWDIR_SUPPORT 
	RAWDIR, 
#endif 
```

File: src/common/Time.h (hunk 1 above)
Comment (bluca) [inline src/common/Time.h]: Done, for all others as well

File: src/common/ipfixlolib/ipfixlolib.c (hunk 6 above)
Comment (ogasser) [inline src/common/ipfixlolib/ipfixlolib.c]: Ok, then it makes sense to handle it like this.

**Comment by bluca** [comment]
Thanks! Next one has important fixes for the behaviour of the flow cache timeouts: https://github.com/tumi8/vermont/pull/60
